packages/react-components/react-storybook-addon-export-to-sandbox/src/decorators/with-export-to-sandbox-button.ts

```diff
@@ -1,15 +1,22 @@
-import { addDemoActionButton } from '../sandbox-factory';
+import { makeDecorator } from '@storybook/preview-api';
+import { useEffect } from 'react';
 
 import { StoryContext } from '../types';
+import { addDemoActionButton } from '../sandbox-factory';
+
+export const withExportToSandboxButton = makeDecorator({
+  name: 'export-to-sandbox-button',
+  parameterName: 'exportToSandbox',
+  wrapper: (getStory, context) => {
```

Comment:
this is becoming a hit n miss, we need to troubleshoot in more detail why those regressions started to happen as it's patch over patch that is not reliable.

- why using deprecated `makeDecorator` vs simple decorator function makes it work ?
- useEffect from react should be avoided in decorators as it makes it non framework agnostic, that;s why previously `@storybook/preview-api` `useEffect` was used